packages/salesforcedx-vscode-core/test/jest/metadataSupport/metadataHoverProvider.test.ts

```diff
@@ -0,0 +1,860 @@
+          // Return the range for the element name only (excluding < and >)
+          const nameStart = matchStart + 1 + (match[1] ? match[1].length : 0); // Skip < and optional /
+          const nameEnd = nameStart + elementName.length;
+          return {
+            start: { line: position.line, character: nameStart },
+            end: { line: position.line, character: nameEnd }
+          } as vscode.Range;
+        }
+      }
+
+      // Fallback to original word finding logic
+      let start = position.character;
+      let end = position.character;
+
+      // Move start backwards to find word beginning
+      while (start > 0 && /[a-zA-Z_]/.test(line[start - 1])) {
+        start--;
+      }
+
+      // Move end forwards to find word end
+      while (end < line.length && /[a-zA-Z_]/.test(line[end])) {
+        end++;
+      }
+
+      // If no word found, return undefined
+      if (start === end) return undefined;
+
+      return {
+        start: { line: position.line, character: start },
+        end: { line: position.line, character: end }
+      } as vscode.Range;
+    }),
+    lineAt: jest.fn((line: number) => ({
+      text: content.split('\n')[line] || '',
+      lineNumber: line
+    }))
+  }) as any;
+
+describe('MetadataHoverProvider', () => {
+  let hoverProvider: MetadataHoverProvider;
+
+  beforeEach(() => {
+    hoverProvider = new MetadataHoverProvider();
+    jest.clearAllMocks();
+  });
+
+  describe('isMetadataFile', () => {
+    it('should identify metadata files correctly', () => {
+      const metadataDoc = createMockDocument('test-meta.xml', '<ApexClass>');
+      const regularDoc = createMockDocument('test.txt', 'some text');
+
+      // Use reflection to access private method for testing
+      const isMetadataFile = (hoverProvider as any).isMetadataFile;
+
+      expect(isMetadataFile.call(hoverProvider, metadataDoc)).toBe(true);
+      expect(isMetadataFile.call(hoverProvider, regularDoc)).toBe(false);
+    });
+
+    it('should identify XML files with metadata namespace', () => {
+      const xmlDoc = createMockDocument(
+        'test.xml',
+        '<?xml version="1.0"?><root xmlns="http://soap.sforce.com/2006/04/metadata">'
+      );
+
+      const isMetadataFile = (hoverProvider as any).isMetadataFile;
+      expect(isMetadataFile.call(hoverProvider, xmlDoc)).toBe(true);
+    });
+  });
+
+  describe('extractMetadataType', () => {
+    it('should extract metadata type from XML element', () => {
+      const extractMetadataType = (hoverProvider as any).extractMetadataType;
+
+      const result = extractMetadataType.call(
+        hoverProvider,
+        '<ApexClass xmlns="http://soap.sforce.com/2006/04/metadata">',
+        'ApexClass',
+        5
+      );
+
+      expect(result).toBe('ApexClass');
+    });
+
+    it('should handle namespaced elements', () => {
+      const extractMetadataType = (hoverProvider as any).extractMetadataType;
```

Comment:
I know it's a test, but `any` opens up cans of worms.  this also didn't need to be a method